pkg/operator/ceph/object/bucket/util.go

```diff
@@ -87,12 +88,25 @@ func (p *Provisioner) getObjectStore() (*cephv1.CephObjectStore, error) {
 	return store, err
 }
 
-func MaxObjectQuota(AdditionalConfig map[string]string) string {
-	return AdditionalConfig["maxObjects"]
-}
+func additionalConfigSpecFromMap(config map[string]string) (*additionalConfigSpec, error) {
+	var err error
+	spec := additionalConfigSpec{}
+
+	if _, ok := config["maxObjects"]; ok {
```

Comment:
For repeated similar operations on a struct, within the rook coding style, is it permissible to use `reflect.Indirect()` to DRY it up?

Reply:
I often find that `reflect` can be at odds with readability, and my gut says `reflect` here might make things overly confusing for junior engineers who need to touch the code in the future. 

I also added a question/suggestion in my review that would change how this code works, so it may (or may not be) moot after that.

In general, I find that some of the conventional wisdom on DRY can be a little overzealous. Maybe some of that has to do specifically with how Golang tends to be written/read. Maintaining Rook, I find copy-pasted code is often easy to read and sufficient when dealing with up to 3 "adjacent things". For code dealing with 4+ things or non-adjacent code, I start to care more about DRY.

Reply:
I dislike duplication of code for more than 2 items.  The question was asked with knowledge "from the future" where there are 4 keys with the same conversion: https://github.com/rook/rook/pull/14834

Reply:
I'd prefer to find a way to DRY it up without `reflect`. Keeping code more accessible to junior engineers (for whom reflect may be akin to magic) is still good practice.